src/profile-logic/process-profile.js

```diff
@@ -854,7 +854,7 @@ function _processMarkerPayload(
   // pre-emptively done for every single marker payload.
   //
   // Warning: This function converts the payload into an any type
-  const payload = _convertStackToCause(geckoPayload);
+  const payload: MarkerPayload_Gecko = _convertStackToCause(geckoPayload);
```

Comment:
previously `payload` was `any` at this point, so I'm adding some more type safety in this function.

Reply:
Hmm, this doesn't look right. Because `MarkerPayload_Gecko` has `stack` and not `cause`. After converting `stack` to `cause`, it's not a gecko payload anymore. Probably something in between gecko and processed payload. I think `any` type suits better here.

Reply:
Mmm that's true, but we're not using the stack in the rest of this function, and we still benefit from the rest of the type related to payloads. I'll change to use `$ReplaceCauseWithStack` that we use in types/markers.js already (possibly in `_convertStackToCause` directly rather than here).